distributed/worker.py

```diff
@@ -1591,40 +1596,13 @@ async def close(  # type: ignore
         self.batched_send({"op": "close-stream"})
 
         if self.batched_stream:
-            with suppress(TimeoutError):
-                await self.batched_stream.close(timedelta(seconds=timeout))
+            await self.batched_stream.close()
 
         for executor in self.executors.values():
             if executor is utils._offload_executor:
                 continue  # Never shutdown the offload executor
 
-            def _close(executor, wait):
-                if isinstance(executor, ThreadPoolExecutor):
-                    executor._work_queue.queue.clear()
-                    executor.shutdown(wait=wait, timeout=timeout)
-                else:
-                    executor.shutdown(wait=wait)
-
-            # Waiting for the shutdown can block the event loop causing
-            # weird deadlocks particularly if the task that is executing in
-            # the thread is waiting for a server reply, e.g. when using
-            # worker clients, semaphores, etc.
-            if is_python_shutting_down():
-                # If we're shutting down there is no need to wait for daemon
-                # threads to finish
-                _close(executor=executor, wait=False)
-            else:
-                try:
-                    await to_thread(_close, executor=executor, wait=executor_wait)
-                except RuntimeError:  # Are we shutting down the process?
-                    logger.error(
-                        "Could not close executor %r by dispatching to thread. Trying synchronously.",
-                        executor,
-                        exc_info=True,
-                    )
-                    _close(
-                        executor=executor, wait=executor_wait
-                    )  # Just run it directly
```

Comment:
This dispatch to a threadpool executor makes all the "close the assert that worker is gone" tests fail because the worker is closing just a tiny bit faster now. In fact, just skipping one tick (sleep(0)) is sufficient to make all other tests pass.